plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -173,101 +167,108 @@ export class PgInstrumentation extends InstrumentationBase {
+        // to properly narrow arg0, but TS 4.3.5 does not.
+        if (instrumentationConfig.addSqlCommenterCommentToQueries) {
+          args[0] = firstArgIsString
+            ? utils.addSqlCommenterComment(span, arg0 as string)
+            : firstArgIsQueryObjectWithText
+            ? {
+                ...(arg0 as utils.ObjectWithText),
+                text: utils.addSqlCommenterComment(
+                  span,
+                  (arg0 as utils.ObjectWithText).text
+                ),
+              }
+            : args[0];
         }
 
-        // Bind callback to parent span
+        // Bind callback (if any) to parent span (if any)
         if (args.length > 0) {
           const parentSpan = trace.getSpan(context.active());
           if (typeof args[args.length - 1] === 'function') {
             // Patch ParameterQuery callback
             args[args.length - 1] = utils.patchCallback(
-              plugin.getConfig(),
+              instrumentationConfig,
               span,
-              args[args.length - 1] as PostgresCallback
+              args[args.length - 1] as PostgresCallback // nb: not type safe.
             );
+
             // If a parent span exists, bind the callback
             if (parentSpan) {
               args[args.length - 1] = context.bind(
                 context.active(),
                 args[args.length - 1]
               );
             }
-          } else if (
-            typeof (args[0] as NormalizedQueryConfig).callback === 'function'
-          ) {
+          } else if (typeof queryConfig?.callback === 'function') {
             // Patch ConfigQuery callback
             let callback = utils.patchCallback(
               plugin.getConfig(),
               span,
-              (args[0] as NormalizedQueryConfig).callback!
+              queryConfig.callback as PostgresCallback // nb: not type safe.
             );
+
             // If a parent span existed, bind the callback
             if (parentSpan) {
               callback = context.bind(context.active(), callback);
             }
 
-            // Copy the callback instead of writing to args.callback so that we don't modify user's
-            // original callback reference
+            // Copy the callback instead of writing to args.callback so that we
+            // don't modify user's original callback reference
             args[0] = { ...(args[0] as object), callback };
           }
         }
 
-        // Perform the original query
-        const result: unknown = original.apply(this, args as any);
+        let result: unknown;
+        try {
+          result = original.apply(this, args as never);
+        } catch (e: unknown) {
+          // span.recordException(e);
+          span.setStatus({
+            code: SpanStatusCode.ERROR,
+            message: utils.getErrorMessage(e),
+          });
+          span.end();
+          throw e;
+        }
```

Comment:
is it feasible to write a test for this case?
Looks like when [first argument to query is `undefined` or `null` will trigger it to throw](https://github.com/brianc/node-postgres/blob/27d612a2ac2df8737397019a5806f745f19b760e/packages/pg/lib/client.js#L507) and it will also assert that instrumentation does not throw itself before calling origianl

Reply:
Yup, I added explicit tests now for `client.query(undefined)` and `client.query(null)`. I think that's what you were talking about? The code on main already had some tests that the error thrown is from the pg library (not the instrumentation) and that a span is generated in the error status. I suppose I could also asserting on the span's message, but I don't think we want to couple the tests to the exact wording of the pg package's current errors.

Reply:
Right, I didn't see it already had tests, sorry about that
I thought the code to capture `original` invocations exceptions is new, but now I see it was just moved from `handleInvalidQuery` function 💯